src/bank00.asm

```diff
@@ -10045,20 +10044,29 @@ addMoney:
     ld   E, A                                          ;; 00:3d7a $5f
     add  HL, DE                                        ;; 00:3d7b $19
     pop  DE                                            ;; 00:3d7c $d1
-    jr   NC, .set                                      ;; 00:3d7d $30 $03
-    ld   HL, rIE                                       ;; 00:3d7f $21 $ff $ff
-.set:
+    push HL
+    ld   HL, wStatusEffect
+    jr   NC, .set_1
+    set  7, [HL]
+.set_1:
+    bit  7, [HL]
+    pop  HL
+    jr   Z, .set_2
+    ld   A, L
+    sub  A, $9f
+    ld   A, H
+    sbc  A, $86
+    jr   C, .set_2
+    ld   HL, $869f
```

Comment:
Took me a bit to figure out that $01869f is 99999. Could use a comment.

Reply:
commented, thanks!